src/braket/circuits/gate.py

```diff
@@ -49,18 +73,35 @@ def to_ir(self, target: QubitSet) -> Any:
         Returns:
             IR object of the quantum operator and target
         """
+        if self._adjoint:
+            # Use adjoint list of original gate
+            return [elem.to_ir(target) for elem in self.adjoint().adjoint_list()]
```

Comment:
`.adjoin().adjoint_list()` doesn't make sense to me from the description of these functions.

Reply:
`adjoint_list` will often contain copies of `self`, and we don't want to recursively call `to_ir` on self because it'll never end.

Reply:
I think the best way to go about this is to throw an exception if `_adjoint` is `True` for any gate in `adjoint_list`, and include in the `adjoint_list` doc that the adjoint list _must_ consist only of non-adjoint gates.